src/rust/src/lib.rs

```diff
@@ -2,7 +2,86 @@
 // 2.0, and the BSD License. See the LICENSE file in the root of this repository
 // for complete details.
 
+/// Returns the value of the input with the most-significant-bit copied to all
+/// of the bits.
+fn duplicate_msb_to_all(a: u8) -> u8 {
+    0u8.wrapping_sub(a >> 7)
+}
+
+/// This returns 0xFF if a < b else 0x00, but does so in a constant time
+/// fashion.
+fn constant_time_lt(a: u8, b: u8) -> u8 {
```

Comment:
any specific reason not to use subtle https://github.com/dalek-cryptography/subtle for constant time operations?

Reply:
Note that there has been some [drama around maintainership](https://old.reddit.com/r/rust/comments/lcsg0u/ownership_and_maintenance_struggles_in/) in dalek-cryptography. There's a fork called [subtle-ng](https://github.com/zkcrypto/subtle-ng) now.